plugins/repository-hdfs/src/test/java/org/elasticsearch/repositories/hdfs/HdfsBlobStoreContainerTests.java

```diff
@@ -18,16 +18,78 @@
  */
 package org.elasticsearch.repositories.hdfs;
 
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.AbstractFileSystem;
 import org.apache.hadoop.fs.FileContext;
+import org.apache.hadoop.fs.UnsupportedFileSystemException;
 import org.elasticsearch.common.blobstore.BlobStore;
 import org.elasticsearch.repositories.ESBlobStoreContainerTestCase;
 
+import javax.security.auth.Subject;
 import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.security.AccessController;
+import java.security.Principal;
+import java.security.PrivilegedAction;
+import java.util.Collections;
 
 public class HdfsBlobStoreContainerTests extends ESBlobStoreContainerTestCase {
 
     @Override
     protected BlobStore newBlobStore() throws IOException {
-        return new HdfsBlobStore(FileContext.getFileContext(), "", 100);
+        Path tempDir = createTempDir(null);
+        FileContext fileContext = AccessController.doPrivileged(
+                new PrivilegedAction<FileContext>() {
+                    @Override
+                    public FileContext run() {
+                        return createContext(tempDir.toUri());
+            }
+        });
+
+        return new HdfsBlobStore(fileContext, "temp", 100);
+    }
+
+    public FileContext createContext(URI uri) {
+        Configuration cfg = new Configuration(true);
+        cfg.setClassLoader(HdfsRepository.class.getClassLoader());
+        cfg.reloadConfiguration();
```

Comment:
Sorry, I don't follow.

The iteration over `conf.` properties is a way for the plugin to enrich the created `Configuration` file (see the plugin documentation for an example).

As for the initial question - what do you mean by "allow it to create"? `Configuration` and `FileContext` are just wrappers.